scip-java/src/main/scala/com/sourcegraph/scip_java/buildtools/ScipBuildTool.scala

```diff
@@ -478,10 +503,38 @@ class ScipBuildTool(index: IndexCommand) extends BuildTool("SCIP", index) {
     arguments += generatedDir(tmp, "s")
     arguments += "-h"
     arguments += generatedDir(tmp, "h")
-    arguments += "-classpath"
-    arguments += actualClasspath.mkString(File.pathSeparator)
+    if (actualClasspath.nonEmpty) {
+      arguments += "-classpath"
+      arguments += actualClasspath.mkString(File.pathSeparator)
+    }
     arguments +=
       s"-Xplugin:semanticdb -targetroot:$targetroot -sourceroot:$sourceroot"
+    if (config.processorpath.nonEmpty) {
+      arguments += "-processorpath"
+      val processorpath =
+        semanticdbJar.toString ::
+          config
+            .processorpath
+            .flatMap(path => guessBazelJar(path, index.workingDirectory))
+            .map(_.toString)
+      arguments += processorpath.mkString(File.pathSeparator)
+    }
+    val isIgnoredAnnotationProcessor =
+      ScipBuildTool.isIgnoredAnnotationProcessor ++
+        index.scipIgnoredAnnotationProcessors
+    val processors = config.processors.filterNot(isIgnoredAnnotationProcessor)
+    if (processors.nonEmpty) {
+      arguments += "-processor"
+      arguments += processors.mkString(",")
+    }
+    arguments ++=
+      fixJavacOptions(config.javacOptions).filterNot(option =>
+        option.startsWith("-Xep") || option.startsWith("-Xplugin:semanticdb") ||
```

Comment:
Of these three I only know `-Xplugin` - what do `-Xep` and `-XD` prefixes mean?

Reply:
I moved this logic into `fixJavacOptions` and added comments for the other ones. `-Xep` is for error prone and I'm not sure what `-XD` is for, but it was failing the build for some reason. This might no longer be needed after I updated the aspect to correctly depend on the sibling target, which fixed a bunch of issues (esp. related to the processorpath).